code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	max_cooldown = 50 
+
+
+/obj/effect/warped_rune/greyspace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/greyspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		slimerevival()
+	
+///Makes a slime of the color of extract that were put on the runes.can only take one type of extract between slime spawning.
+/obj/effect/warped_rune/greyspace/proc/slimerevival()
+	if(i <! 8) // this shouldn't happen and will bring back the count to 7
+		i = 7
+		return
+
+	for(var/obj/item/slime_extract/extr in T)
+		if( extr.color_slime != extractype && extractype != 0) //check if the extract is the first one or of the right color.
+			return
+		extractype = extr.color_slime  //keep the slime extract color in storage
+		qdel(extr)    //vores the slime extract
+		playsound(T, 'sound/effects/splat.ogg', 20, TRUE)
+		i++
+		if (i != 8)
+			return
+
+		playsound(T, 'sound/effects/splat.ogg', 20, TRUE)
+		S = new(T, extr.color_slime)  //spawn a slime from the extract's color (I'm keeping S as a variable just in case I need to change the slime)
+		i = 0
+		extractype = 0 // reset the type to allow a new extract type
+
+
+
+/*The orange rune warp basically ignites whoever walks on it,the fire will teleport you at random as long as you are on fire*/
+
+
+/obj/item/slimecross/warping/orange
+	desc = "Creates a rune "
+	colour = "orange"
+	runepath = /obj/effect/warped_rune/orangespace
+	effect_desc = "Creates a rune burning with bluespace fire, anyone walking into the rune will ignite and teleport randomly as long as they are on fire"
+
+/obj/effect/warped_rune/orangespace
+	name = "orange space rune"
+	desc = "When all is reduced to ash, it shall be reborn from the depth of bluespace"
+	icon_state = "bluespace_fire"
+	max_cooldown = 50
+
+
+
+/obj/effect/warped_rune/orangespace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/teleport_fire) //TODO : figure out how to use the shortened version of this
+	START_PROCESSING(SSprocessing, src)
+
+/obj/effect/warped_rune/orangespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		teleport_fire() //triggers it every 5 seconds anyway
+
+
+
+
+///teleport people and put them on fire if they run into the rune.
+/obj/effect/warped_rune/orangespace/proc/teleport_fire()
+	if(!locate(/obj/effect/hotspot) in T) //doesn't teleport item but put them on fire anyway for good measure.
+		new /obj/effect/hotspot(T)
+
+	for(var/mob/living/F in T)
+		do_teleport(F, T, 3, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)//only set the teleport to one for now
+		F.adjust_fire_stacks(10)
+		F.IgniteMob()
+
+/*
+		for(F.fire_stacks > 0 && !istype(F,/mob/living/simple_animal)) //Keeps teleporting the guy as long as he is on fire (doesn't include simplemobs) WIP
+			do_teleport(F, get_turf(F), 5, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
```

Comment:
there's no delay for this, so people who get teleported by this rune are gonna get teleported like 10+ times per second

also, this rune won't set anyone else who was on the rune on fire or teleport them until the people before them in the queue have been extinguished.

use, like, callbacks or something for this